### ISSUE TITLE
How do I know that the upgrade of the NVIDIA device plugin went well?

### ISSUE THREAD

**Issue by Borchies** [NVIDIA/k8s-device-plugin#349]
_The template below is mostly useful for bug reports and support questions. Feel free to remove anything which doesn't apply to you and add more information where it makes sense._
---

### 1. Issue or feature description

Hello! More than a bug report, this is just a question that I have. I recently updated the device plugin from `0.12.2` to `0.12.3`.

All I did was to modify the container's image from `nvcr.io/nvidia/k8s-device-plugin:v0.12.2` to `nvcr.io/nvidia/k8s-device-plugin:v0.12.3` and nothing else.

Then, I ran `nvidia-smi` in the node that has the pod of the NVIDIA device plugin and I got the following outcome:
```+-----------------------------------------------------------------------------+
| NVIDIA-SMI 470.57.02    Driver Version: 470.57.02    CUDA Version: 11.4     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla T4            On   | 00000000:00:1E.0 Off |                    0 |
| N/A   51C    P0    34W /  70W |   8682MiB / 15109MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      9430      C   python3                          8679MiB |
+-----------------------------------------------------------------------------+
```

As you can see, CUDA version is still `11.4`, so my questions are:
- Shouldn't it be updated to `11.7.1`? 
- Is there something else I need to do apart from upgrading the image's version of the container? 
- What changes should I expect to see in the output of `nvidia-smi` command?

**Comment by elezar** [comment]
Is this run on the node itself or in one of the containers of the device plugin pod?

The device plugin does not modify the driver installation at all. The CUDA version mentioned in the release notes only refers to the tag of the CUDA base image that we use for the container. This should not affect functionality at all and is only updated for licensing reasons and to ensure that the base image is updated to address as many CVEs as possible at the time of release.

**Comment by Borchies** [comment]
I see, so nothing should be different in the `nvidia-smi` outcome, right?

What we have is a Daemonset that deploys the pod that has the `nvcr.io/nvidia/k8s-device-plugin:v0.12.3` image in the container.

**Comment by elezar** [comment]
@Borchies yes, that is correct. The nvidia-smi output should stay the same across Device Plugin upgrades. It sounds as if you're using the driver installed on the host -- in which case your `nvidia-smi` output should only change if this driver is updated.

**Comment by Borchies** [comment]
Understood, thanks for the clarification! I will close this issue now. Have a good day!
